skywalking-collector/skywalking-collector-worker/src/main/java/com/a/eye/skywalking/collector/worker/storage/WindowData.java

```diff
@@ -0,0 +1,52 @@
+package com.a.eye.skywalking.collector.worker.storage;
+
+import java.util.Map;
+
+/**
+ * @author pengys5
+ */
+public class WindowData<T extends Data> {
+    private Map<String, T> data;
+    private boolean isHold;
```

Comment:
You are processing data across-thread, please set `volatile` key word for your fields.